src/components/PrometheusGraph/PrometheusGraph.js

```diff
@@ -351,18 +360,58 @@ class PrometheusGraph extends React.PureComponent {
       })),
     }));
 
-    this.setState({ multiSeries });
+    const visibleMultiSeries = // If no series is selected, show all of them.
+      this.state.selectedLegendKeys.length === 0
+        ? multiSeries
+        : // Otherwise show only the selected multi series.
+          multiSeries.filter(series =>
+            this.state.selectedLegendKeys.includes(series.key)
+          );
+
+    const { timeScale, valueScale } = this.prepareTimeAndValueScales(
+      props,
+      visibleMultiSeries
+    );
+
+    this.setState({
+      multiSeries,
+      visibleMultiSeries,
+      timeScale,
+      valueScale,
+    });
+  };
+
+  prepareTimeAndValueScales = (props, multiseries) => {
+    let chartWidth;
+    let chartHeight;
+
+    if (this.chartRef) {
+      ({
+        width: chartWidth,
+        height: chartHeight,
+      } = this.chartRef.getSvgBoundingRect());
```

Comment:
What's with the `()` wrapper? :D

Reply:
Necessary for destructuring into variables that are already declared. I had to google that one 😄